src/parser.rs

```diff
@@ -263,4 +279,12 @@ mod tests {
         assert_eq!(rest, "");
         // TODO: Assert AST result
     }
+
+    #[test]
+    fn shape() {
+        let expression = "z: (1 * (2.0 + 5 / (4 - 2)))\n square x\nsquare x+(13.2) 9.2\n circle x+23.9\n circle z\n circle (12.93*(2+(9-7.6/129.92)))\n circle";
+        let (rest, _ast) = parser(expression).unwrap();
+
+        assert_eq!(rest, "");
```

Comment:
Make some asserts that actually tests everything is coming out correctly from the parser.
It is good that the parser does not fail, but you should also make sure that the generated AST is what you expect it to be.

Instead of a single test with all the possible declarations, write 4 simpler tests that check:
- `square`
- `square 1`
- `square 1 3`
- `x: 1\n square x x + 3`